tests/test_generate_diff.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+from gendiff.scripts.gendiff import generate_diff
+
+
+@pytest.fixture
+def file1() -> list:
+    file1 = "tests/fixtures/file1.json"
+    file1_empty = "tests/fixtures/file1_empty.json"
+    return [file1, file1_empty]
+
+
+@pytest.fixture
+def file2() -> list:
+    file2 = "tests/fixtures/file2.json"
+    file2_empty = "tests/fixtures/file1_empty.json"
+    return [file2, file2_empty]
+
+
+@pytest.fixture
+def diff() -> list:
+    diff_file_list = [
+        "tests/fixtures/diff.txt",
+        "tests/fixtures/diff_f1_empty.txt",
+        "tests/fixtures/diff_f2_empty.txt",
+    ]
+    diff_list = []
+    for diff_file in diff_file_list:
+        with open(diff_file, "r") as diff:
+            diff_str = ""
+            for line in diff:
+                diff_str += line
+        diff_list.append(diff_str)
+    return diff_list
+
+
+def test_generate_diff(file1: list, file2: list, diff: list) -> None:
+    assert generate_diff(file1[0], file2[0]) == diff[0]
+
+
+def test_generate_diff_both_empty(file1: list, file2: list) -> None:
+    assert generate_diff(file1[1], file2[1]) == "{}"
+
+
+def test_generate_diff_f1_empty(file1: list, file2: list, diff: list) -> None:
+    assert generate_diff(file1[1], file2[0]) == diff[1]
+
+
+def test_generate_diff_f2_empty(file1: list, file2: list, diff: list) -> None:
+    assert generate_diff(file1[0], file2[1]) == diff[2]
```

Comment:
too hard to read :(  

it's good if test has properly defined input, action and output (comparison to expected)